mfs/system.go

```diff
@@ -123,6 +124,24 @@ func (kr *Root) Flush() error {
 	return nil
 }
 
+// PinSource returns information about pinning requirements.
+func (kr *Root) PinSource() *pin.Source {
+	return &pin.Source{
+		Get: func() ([]*cid.Cid, error) {
+			err := kr.Flush()
```

Comment:
I assume this won't trigger a recursive GC. We take a lock, right? However, this can cause us to go over the GC limit even more. We should probably document the fact that the "max repo size" isn't *really* the max.

Reply:
We can also get in a bad state here where we can't flush (no disk space) and can't GC (can't flush). Not sure if we can do something about it now but...

Reply:
There isn't much we can do here to recover I think, I need the actual state of the tree to perform the GC without data loss and I can't get it unless I flush the tree so it propagates the DAG updates.